### ISSUE TITLE
PK-Sim database: unnecessary foreign key?

### ISSUE THREAD

**Issue by Yuri05** [Open-Systems-Pharmacology/PK-Sim#2856]
In the PK-Sim database we have the FK relationship between **tab_event_changed_XXX** and **tab_event_conditions** which includes the **calculation_method**:
![grafik](https://github.com/Open-Systems-Pharmacology/PK-Sim/assets/25061876/db201b46-d6ac-4c13-bea4-556e82cea221)

While other columns in this FK-relationship make perfectly sense for me: I don't see why the calculation method should be the same.
(with the current FK: "_the calculation method of the new assigned formula must be the same as the calculation method of the event condition formula"_).

Because of this, we need to duplicate some formulas (e.g. when resetting a parameter to the original (fasted state) formula after a meal: we cannot reuse the old formula, because it has a different calculation method ("Lumen_PKSim") and must redefine the formula with the calculation method "Events"
![grafik](https://github.com/Open-Systems-Pharmacology/PK-Sim/assets/25061876/e0738a5f-0905-4e7a-889f-d3d5deaa99bf)

So I suggest to remove the **calculation_method** column from this FK

**Comment by Yuri05** [comment]
@msevestre Any concerns?

**Comment by msevestre** [comment]
nope. It's probably just because we always create foreign key based on name and we did not think about it

**Comment by Yuri05** [comment]
Note for self: need to add unique index in **tab_event_conditions** on (**event_id**, **event_container_type**, **event_name**) for that.
